conn.go

```diff
@@ -62,14 +62,16 @@ func (c *otConn) Ping(ctx context.Context) (err error) {
 	}()
 
 	if c.cfg.SpanOptions.Ping {
-		var span trace.Span
-		ctx, span = createSpan(ctx, c.cfg, method, false, "", nil)
-		defer func() {
-			if err != nil {
-				recordSpanError(span, c.cfg.SpanOptions, err)
-			}
-			span.End()
-		}()
+		if !omitSpan(ctx, c.cfg.SpanOptions, method, "", nil) {
```

Comment:
Should we change this function name to `filterSpan` or `spanFilter`? Returning true is quite clear then returning false and using `!`.

The implementation of this function could be:

```go
return spanOptions.SpanFilter == nil || spanOptions.SpanFilter(ctx, method, query, args)
```

---

Though it is my preference, it is up to you to change it.

Reply:
I've updated this to filterSpan.